### PR TITLE
VSF2-60 - Feature/wishlist tests

### PR DESCRIPTION
**Pull request by Wojdylak**

### CODE REVIEW

File: spec/CommandHandler/Cart/AddItemsToCartHandlerSpec.php [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+use Sylius\Component\Core\Model\OrderInterface;
+use Sylius\Component\Core\Model\OrderItemInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Core\Repository\OrderRepositoryInterface;
+use Sylius\Component\Inventory\Checker\AvailabilityCheckerInterface;
+use Sylius\Component\Order\Modifier\OrderItemQuantityModifierInterface;
+use Sylius\Component\Order\Modifier\OrderModifierInterface;
+use Webmozart\Assert\InvalidArgumentException;
+
+final class AddItemsToCartHandlerSpec extends ObjectBehavior
```
Comment (senghe) [inline spec/CommandHandler/Cart/AddItemsToCartHandlerSpec.php:28]: At least one missing scenario :vulcan_salute:

File: src/Doctrine/Orm/Extension/WishlistsCurrentUserExtension.php [hunk 2]
```diff
@@ -63,7 +63,9 @@ public function applyToCollection(
         }
 
         $rootAlias = $queryBuilder->getRootAliases()[0];
-        $queryBuilder->andWhere(sprintf('%s.shopUser = :current_user', $rootAlias));
-        $queryBuilder->setParameter('current_user', $user->getId());
+        $queryBuilder->andWhere(
```
Comment (senghe) [inline src/Doctrine/Orm/Extension/WishlistsCurrentUserExtension.php:66]: I would use PHPUnit integration tests nor PHPSpec ones

Comment (senghe) [reply src/Doctrine/Orm/Extension/WishlistsCurrentUserExtension.php:66]: lchrusciel/api-test-case

**Review by senghe — CHANGES_REQUESTED** [review]

File: features/shop/cart/adding_products_to_cart.feature [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    In order buy a products
+    I need to be able to add multiple products to cart
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And the store has a product "Harness climbing" priced at "$30.00"
+        And the store has a product "Rope" priced at "$40.00"
+
+    @graphql
+    Scenario: Placing order as quest
```
Comment (senghe) [inline features/shop/cart/adding_products_to_cart.feature:12]:
```suggestion
    Scenario: Placing order as guest
```

File: features/shop/cart/adding_products_to_cart.feature [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+            | order.items.edges.0.node.productName  | Harness climbing  | string    |
+            | order.items.edges.0.node.quantity     | 2                 | int       |
+            | order.items.edges.1.node.productName  | Rope              | string    |
+            | order.items.edges.1.node.quantity     | 3                 | int       |
+            | order.total                           | 18000             | int       |
+
+    @graphql
+    Scenario: Placing order as customer
+        Given there is a customer "Adam Ondra" identified by an email "aondra@climb.com" and a password "ardno1"
+        And I create a JWT Token for customer identified by an email "aondra@climb.com"
```
Comment (senghe) [inline features/shop/cart/adding_products_to_cart.feature:34]: And I am authorized as "aondra@climb.com"

File: features/shop/wishlist/customer/clearing_wishlist.feature [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        And the store has a product "Rope" priced at "$40.00"
+        And there is a customer "Alex Honnold" identified by an email "ahonnold@climb.com" and a password "dlonnoh1"
+        And user "ahonnold@climb.com" has a wishlist named "For Alex"
+        And there is a customer "Adam Ondra" identified by an email "aondra@climb.com" and a password "ardno1"
+        And user "aondra@climb.com" has a wishlist named "For me"
+        And I create a JWT Token for customer identified by an email "aondra@climb.com"
+
+    @graphql
+    Scenario: Cleaning a wishlist
+        Given user have a product "ROPE" in my wishlist "For me"
```
Comment (senghe) [inline features/shop/wishlist/customer/clearing_wishlist.feature:19]: user has

Comment (senghe) [reply features/shop/wishlist/customer/clearing_wishlist.feature:19]: You have the literal issue in a few places. Could you localize them and fix also?

File: features/shop/wishlist/customer/clearing_wishlist.feature [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+        And user have a product "HARNESS_CLIMBING" in my wishlist "For me"
+        And There is operation to clear wishlist
+        And this operation has "id" variable with iri value of object "For me"
+        When I send that GraphQL request as authorised user
+        Then I should receive a JSON response
+        And This response should contain "wishlist.wishlistProducts.totalCount" equal to 0 as "int"
+
+    @graphql
+    Scenario: Cleaning a wishlist
```
Comment (senghe) [inline features/shop/wishlist/customer/clearing_wishlist.feature:28]: Cleaning an empty wishlist ?

File: features/shop/wishlist/customer/creating_wishlist.feature [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+    In order to create a new wishlist
+    As a customer
+    I want to be able to create new wishlists
+
+    Background:
+        Given the store operates on a single channel in "United States"
+        And there is a customer "Alex Honnold" identified by an email "ahonnold@climb.com" and a password "dlonnoh1"
+        And user "ahonnold@climb.com" has a wishlist named "For me"
+        And there is a customer "Adam Ondra" identified by an email "aondra@climb.com" and a password "ardno1"
+        And I create a JWT Token for customer identified by an email "aondra@climb.com"
```
Comment (senghe) [inline features/shop/wishlist/customer/creating_wishlist.feature:12]: I authorize as "aondra@climb.com"

File: features/shop/wishlist/customer/adding_product_to_wishlist.feature [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+        Given There is operation to add product to wishlist
+        And this operation has "id" variable with iri value of object "For me"
+        And this operation has "productVariant" variable with iri product variant "HARNESS_CLIMBING"
+        When I send that GraphQL request as authorised user
+        Then I should receive a JSON response
+        And This response body should contain:
+            | key                                                   | value             | type      |
+            | wishlist.name                                         | For me            | string    |
+            | wishlist.wishlistProducts.totalCount                  | 1                 | int       |
+            | wishlist.wishlistProducts.edges.0.node.variant.name   | Harness climbing  | string    |
```
Comment (senghe) [inline features/shop/wishlist/customer/adding_product_to_wishlist.feature:28]: I think the response tables you can change to sentences i.e.: "And the total count of products is 1", "And the first variant in wishlist is Harness climbing".

File: features/shop/wishlist/customer/removing_product_from_wishlist.feature [hunk 9]
```diff
@@ -0,0 +1,54 @@
+@wishlist_as_customer
+Feature: Adding a product to a wishlist
```
Comment (senghe) [inline features/shop/wishlist/customer/removing_product_from_wishlist.feature:2]: Removing product from wishlist

File: features/shop/wishlist/customer/removing_wishlist.feature [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+    Background:
+        Given the store operates on a single channel in "United States"
+        And there is a customer "Alex Honnold" identified by an email "ahonnold@climb.com" and a password "dlonnoh1"
+        And user "ahonnold@climb.com" has a wishlist named "For Alex"
+        And there is a customer "Adam Ondra" identified by an email "aondra@climb.com" and a password "ardno1"
+        And user "aondra@climb.com" has a wishlist named "For me"
+        And I create a JWT Token for customer identified by an email "aondra@climb.com"
+
+    @graphql
+    Scenario: Removing wishlists
```
Comment (senghe) [inline features/shop/wishlist/customer/removing_wishlist.feature:16]: Duplicated scenario name

File: features/shop/wishlist/customer/removing_wishlist.feature [hunk 11]
```diff
@@ -0,0 +1,38 @@
...
+        Given There is operation to remove wishlist
+        And this operation has "id" variable with iri value of object "For me"
+        When I send that GraphQL request as authorised user
+        Then I should receive a JSON response
+        And user "aondra@climb.com" should have 0 wishlists
+
+    @graphql
+    Scenario: Removing wishlists
```
Comment (senghe) [inline features/shop/wishlist/customer/removing_wishlist.feature:24]: How about removing wishlist having products inside?

File: tests/Behat/Resources/suites/cart/place_order_as_customer.yaml [hunk 12]
```diff
@@ -19,4 +19,4 @@ default:
                 - sylius.behat.context.transform.lexical
                 - sylius.behat.context.transform.shared_storage
             filters:
-                tags: "@place_order_customer && @graphql"
+                tags: "@place_order_customer &@graphql"
```
Comment (senghe) [inline tests/Behat/Resources/suites/cart/place_order_as_customer.yaml:22]: possible issue

File: tests/Integration/Doctrine/WishlistsCurrentUserExtensionTest.php [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+        ;
+
+        $queryBuilder = $this->createMock(QueryBuilder::class);
+        $queryBuilder
+            ->expects(self::never())
+            ->method('andWhere')
+        ;
+        $queryNameGenerator = $this->createMock(LegacyQueryNameGeneratorInterface::class);
+
+        $productVariantCurrentVendorExtension = new WishlistsCurrentUserExtension($security);
```
Comment (senghe) [inline tests/Integration/Doctrine/WishlistsCurrentUserExtensionTest.php:40]: You can get the service directly from container

**Review by senghe — CHANGES_REQUESTED** [review]

File: features/shop/wishlist/customer/removing_wishlist.feature (hunk 11 above)
Comment (Wojdylak) [reply features/shop/wishlist/customer/removing_wishlist.feature:24]: I added

File: tests/Behat/Context/GraphqlApiPlatformContext.php [hunk 14]
```diff
@@ -166,6 +196,16 @@ public function thatResponseShouldContainKeyWithValue(
...
+    /**
+     * @Then This response body should contains:
+     */
+    public function thisResponseBodyShouldContain(TableNode $table): void
+    {
+        foreach ($table as $row) {
+            $this->thatResponseShouldContainKeyWithValue($row['key'], $row['value'], $row['type']);
+        }
+    }
+
```
Comment (senghe) [inline tests/Behat/Context/GraphqlApiPlatformContext.php:208]:
```suggestion
```

**Review by senghe — APPROVED** [review]

**Pull request merged**